graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -350,18 +357,136 @@ def run(self) -> None:
+                },
+                "edges_features": {...}
+            }
+
+        Returns
+        -------
+        dict
+            A copy of the ``gsp_config_dict`` modified to be merged with
+            the transformation representations.
+        """
+        gsp_config_dict_copy = copy.deepcopy(gsp_config_dict)
+
+        edge_transformations: Mapping[str, Mapping[str, Mapping]] = transformations["edge_features"]
+        node_transformations: Mapping[str, Mapping[str, Mapping]] = transformations["node_features"]
+        edge_input_dicts: list[dict] = gsp_config_dict_copy["edges"]
+        node_input_dicts: list[dict] = gsp_config_dict_copy["nodes"]
+
+        def get_structure_type(single_input_dict: dict, structure_type: str) -> str:
+            """Gets the node or edge type name from input dict."""
+            if structure_type == "node":
+                type_name = single_input_dict["type"]
+            elif structure_type == "edge":
+                src_type = single_input_dict["source"]["type"]
+                dst_type = single_input_dict["dest"]["type"]
+                relation = single_input_dict["relation"]["type"]
+                type_name = f"{src_type}:{relation}:{dst_type}"
+            else:
+                raise ValueError(
+                    f"Invalid structure type: {structure_type}. Expected 'node' or 'edge'."
+                )
+
+            return type_name
+
+        def append_transformations(
+            structure_input_dicts: list[dict],
+            structure_transformations: Mapping[str, Mapping[str, Mapping]],
+            structure_type: str,
+        ):
+            """Appends the pre-computed transformations to the input dicts."""
+            assert structure_type in ["edge", "node"]
+            for input_dict in structure_input_dicts:
+                type_name = get_structure_type(input_dict, structure_type)
+                # If we have pre-computed transformations for this type
+                if type_name in structure_transformations:
+                    type_transformations: Mapping[str, Mapping] = structure_transformations[
+                        type_name
+                    ]
+                    assert (
+                        "features" in input_dict
+                    ), f"Expected type {type_name} to have have features in the input config"
+
+                    # Iterate over every feature of the type,
+                    # and append representation if one exists
+                    for type_feat_dict in input_dict["features"]:
+                        feat_name = (
+                            type_feat_dict["name"]
+                            if "name" in type_feat_dict
+                            else type_feat_dict["column"]
+                        )
+                        if feat_name in type_transformations:
+                            type_feat_dict["precomputed_transformation"] = type_transformations[
+                                feat_name
```

Comment:
The indent is weird.

Reply:
Answered above, formatting is automatic, we shortened the var names to make it a bit better.